drape_frontend/rule_drawer.cpp

```diff
@@ -292,12 +292,16 @@ void RuleDrawer::ProcessAreaAndPointStyle(FeatureType & f, Stylist const & s, TI
     applyPointStyle = m_globalRect.IsPointInside(featureCenter);
   }
 
+  bool const skipTriangles = isBuildingOutline && m_context->Is3dBuildingsEnabled();
+  if (!skipTriangles && isBuilding && f.GetTrianglesCount(m_zoomLevel) >= 10000)
```

Comment:
It looks like an unobvious hack with many conditions. At least a clarifying comment is needed.

Should this issue be fixed in the generator instead, to avoid any runtime penalty caused by wrong tags?

Reply:
This is the first impression looking at 4 lines diff :)
Reading the whole function context makes it very clear.

Reply:
"Generator fix will follow later" in description

Reply:
Ok, if this hack will be reversed after the generator fix.